src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,129 @@
 package core.basesyntax;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final int INCREASING_NUMBER = 2;
+    private static final float DEFAULT_LOADFACTOR = 0.75f;
+    private int size;
+    private float loadFactor;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        loadFactor = DEFAULT_LOADFACTOR;
+    }
+
+    public MyHashMap(int capacity, float loadFactor) {
+        table = new Node[capacity];
+        this.loadFactor = loadFactor;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int hashPosition = getHashPosition(key);
+        if (size < getThreshold()) {
+            if (table[hashPosition] == null) {
+                table[hashPosition] = new Node<>(key, value, null);
+                ++size;
+            } else {
+                Node<K, V> node = table[hashPosition];
+                while (node != null) {
+                    if (checkNodeForSameKey(node, key)) {
+                        node.value = value;
+                        return;
+                    }
+                    if (node.next == null) {
+                        break;
+                    } else {
+                        node = node.next;
+                    }
+                }
+                node.next = new Node<>(key, value, null);
+                size++;
+            }
+        } else {
+            resize();
+            put(key, value);
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int hashPosition = getHashPosition(key);
+        Node<K, V> node = table[hashPosition];
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private boolean checkNodeForSameKey(Node<K, V> node, K key) {
+        if (Objects.equals(node.key, key)) {
+            return true;
+        } else {
+            return false;
+        }
```

Comment:
we already have required boolean value :)
```suggestion
        return Objects.equals(node.key, key);
```
and not sure after these changes we still need separate method, maybe simpler to replace checkNodeForSameKey with Objects.equals?